### PR TITLE
New python package: Drishti

### PR DESCRIPTION
**Pull request by jeanbez**
Include Drishti, a command-line tool to guide end-users in optimizing I/O in their applications by detecting typical I/O performance pitfalls and providing a set of recommendations.

### CODE REVIEW

**Review by sbyna — DISMISSED** [review]

File: var/spack/repos/builtin/packages/drishti/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    by detecting typical I/O performance pitfalls and providing a set of recommendations.
+    """
+
+    homepage = "https://github.com/hpc-io/drishti"
+    git = "https://github.com/hpc-io/drishti"
+    pypi = "drishti-io/drishti-io-0.4.tar.gz"
+
+    maintainers = ["jeanbez", "sbyna"]
+
+    version("latest", branch="master")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/drishti/package.py]: Why not use `master` as the name of the version?  It is "standard" and would be clearer to people wanting to build based on the branch.

File: var/spack/repos/builtin/packages/drishti/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    version("0.4", sha256="bbbb272b4f6f44ae762f6cba28a2c589e15608691c559af0cc2f552590335d7b")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/drishti/package.py:23]: Confirmed sha256.

File: var/spack/repos/builtin/packages/drishti/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/drishti/package.py]: Confirmed the branch exists.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256 but have a question about the branch version name.

**Comment by tldahlgren** [comment]
Including @adamjstewart since he's the python package guru.

**Comment by tldahlgren** [comment]
Also, to make things easier for reviewers, we ask that future PRs be submitted in non-`develop` branches (see the docs at https://spack.readthedocs.io/en/latest/contribution_guide.html#branching).

File: var/spack/repos/builtin/packages/drishti/package.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    depends_on("darshan-util", type=("run"))
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-pandas", type=("build", "run"))
+    depends_on("py-rich", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/drishti/package.py]:
```suggestion
    depends_on("py-rich@12.5.1", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/drishti/package.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    git = "https://github.com/hpc-io/drishti"
+    pypi = "drishti-io/drishti-io-0.4.tar.gz"
+
+    maintainers = ["jeanbez", "sbyna"]
+
+    version("latest", branch="master")
+
+    version("0.4", sha256="bbbb272b4f6f44ae762f6cba28a2c589e15608691c559af0cc2f552590335d7b")
+
+    depends_on("darshan-util", type=("run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/drishti/package.py:25]: Does this provide the same files as https://pypi.org/project/darshan/, or should we create a `py-darshan` package?

Comment (jeanbez) [reply var/spack/repos/builtin/packages/drishti/package.py:25]: No, actually, we need that package as well. I created it and opened PR #33430. Once that gets merged, I'll update this recipe accordingly.

File: var/spack/repos/builtin/packages/drishti/package.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    by detecting typical I/O performance pitfalls and providing a set of recommendations.
+    """
+
+    homepage = "https://github.com/hpc-io/drishti"
+    git = "https://github.com/hpc-io/drishti"
+    pypi = "drishti-io/drishti-io-0.4.tar.gz"
+
+    maintainers = ["jeanbez", "sbyna"]
+
+    version("latest", branch="master")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/drishti/package.py]:
```suggestion
    version("master", branch="master")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Review by sbyna — DISMISSED** [review]
looks good.

**Review by tldahlgren — DISMISSED** [review]

**Review by alalazo — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**